### PR TITLE
[DXCDT-110] [DXCDT-113] Refactor actions tests

### PR DESCRIPTION
**Pull request by sergiughf**
## Description

Refactor actions tests so they can be ran in any order and they clean up the test tenant after running. 

## Testing

- [x] This change adds test coverage for new/changed/fixed functionality

## Checklist

- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
- [x] I have reviewed my own code beforehand.
- [x] I have added documentation for new/changed functionality in this PR.
- [x] All active GitHub checks for tests, formatting, and security are passing.
- [x] The correct base branch is being used, if not `main`.

### CODE REVIEW

File: management/actions_test.go [hunk 1]
```diff
@@ -53,175 +35,241 @@ func TestActions(t *testing.T) {
...
-				},
-				DisplayName: auth0.String("My test action Binding"),
+	err := m.Action.Delete(expectedAction.GetID())
+
+	assert.NoError(t, err)
+
+	actualAction, err := m.Action.Read(expectedAction.GetID())
+
+	assert.Empty(t, actualAction)
+	assert.EqualError(t, err, "404 Not Found: That action does not exist.")
```
Comment (Widcket) [inline management/actions_test.go:86]: Any way we can assert by the status code (404), so the test doesn't break in the event that the error description changes?

File: management/actions_test.go [hunk 2]
```diff
@@ -53,175 +35,241 @@ func TestActions(t *testing.T) {
...
-			// Expect a 404 as we can't get execution ID via API
-			t.Log(err)
-			return
-		}
-		t.Fatal(errors.New("read execution unexpectedly succeeded"))
-	})
+func TestActionManager_Execution(t *testing.T) {
+	_, err := m.Action.Execution("M9IqRp9wQLaYNrSwz6YPTTIwMjEwNDA0")
+	// Expect a 404 as we can't get execution ID via API
+	assert.EqualError(t, err, "404 Not Found: That execution does not exist.")
```
Comment (Widcket) [inline management/actions_test.go:216]: Same, can we assert by the status code (404) instead?

File: management/actions_test.go (hunk 1 above)
Comment (sergiughf) [reply management/actions_test.go]: Yep, good idea 👍🏻  [954a359](https://github.com/auth0/go-auth0/pull/49/commits/954a3599e0b5ea0f018239b53bd9af5f261bac8f)

File: management/actions_test.go (hunk 2 above)
Comment (sergiughf) [reply management/actions_test.go]: Yep, good idea 👍🏻  [954a359](https://github.com/auth0/go-auth0/pull/49/commits/954a3599e0b5ea0f018239b53bd9af5f261bac8f)

**Pull request merged**